GameOfLife/Bipeds.py

```diff
@@ -27,11 +27,35 @@ def filter_gender(self, value):
         fgend_list.sort()
         return fgend_list
 
-    def filter_generation(self, value):
-        """
-        filter_generation must go through all humans in population and make a new list of humans to return if they match the given generation.
-        It has one generation
+    def filter_generation(self, gen):
         """
+             filter_generation must go through all humans in population and make a new list of humans to return if they match the given generation.
+             It has one generation
+             """
+
+        #key, value
+        gen = gen.lower()
+        gen_list = []
+        gen_mapping = {
+                'genx': range(1965, 1980),
+                'genz': range(1997, 2012),
+                'gena': range(2010, 2020),
+                'mill': range(1981, 1996)
+            }
+        for key, r in gen_mapping.items():
+            if key in gen.lower():

```

Comment:
It would be better to bring this outside of your for loop. 
for example:
```
if gen not in gen_mapping:
    return []
```
Doing this would mean that you checked that the gen is available inside the mapping, and then you define the range you are looking to use as a filter.
`gen_range = gen_mapping[gen]`
Then with a for loop filter check the humans that fit the range. 
Overall the code is much better.